state/address.go

```diff
@@ -278,13 +278,21 @@ func (st *State) apiHostPortsForCAAS(public bool) (addresses []network.SpaceHost
 		logger.Debugf("getting api hostports for CAAS: public %t, addresses %v", public, addresses)
 	}()
 
-	controllerConfig, err := st.ControllerConfig()
+	// We are fetching info about the controller cloud service,
+	// so need the controller state.
+	ctrlSt, err := st.newStateNoWorkers(st.ControllerModelUUID())
```

Comment:
Oops. I missed that we were doing more than accessing the `controllers` collection, which is global.